File: app/lib/features/auth/domain/repositories/auth_repository.dart
import '../entities/auth_session.dart';

abstract class AuthRepository {
  Future<AuthSession> login({
    required String email,
    required String password,
  });

  Future<AuthSession> register({
    required String firstName,
    required String lastName,
    required String email,
    required String password,
    required String role,
    String? gymId,
    String? gymName,
    String? gymSlug,
    String? phone,
  });
}

File: panel/src/features/auth/domain/repositories/AuthRepository.ts
import type { AuthSession } from '../entities/AuthSession';
import type { AuthUser } from '../entities/AuthUser';

export abstract class AuthRepository {
  abstract login(input: {
    email: string;
    password: string;
  }): Promise<AuthSession>;

  abstract register(input: {
    firstName: string;
    lastName: string;
    email: string;
    password: string;
    role: string;
    phone?: string;
    gymId?: string;
    gymName?: string;
    gymSlug?: string;
  }): Promise<AuthSession>;

  abstract me(token: string): Promise<AuthUser>;
}